core/new-gui/src/app/workspace/component/result-panel/result-panel.component.ts

```diff
@@ -125,6 +59,12 @@ export class ResultPanelComponent {
     if (event.current.state === ExecutionState.BreakpointTriggered) {
       return true;
     }
+
+    // transitioning from any state to completed state
+    if (event.current.state === ExecutionState.Completed) {
```

Comment:
I've removed this completely, agreed with your comment about "blink". However, upon testing, this "blink" is actually not happening. I think it is due to when the `this.component` is not being changed, it will not destruct and construct again. So there is no blink effect.